model/custom_field_v2.go

```diff
@@ -0,0 +1,55 @@
+package model
+
+import "github.com/incident-io/singer-tap/client"
+
+type customFieldV2 struct{}
+
+var CustomFieldV2 customFieldV2
+
+func (customFieldV2) Schema() Property {
+	return Property{
+		Types: []string{"object"},
+		Properties: map[string]Property{
+			"id": {
+				Types: []string{"string"},
+			},
+			"catalog_type_id": {
+				Types: []string{"string", "null"},
+			},
+			"name": {
+				Types: []string{"string"},
+			},
+			"field_type": {
+				Types: []string{"string"},
+			},
+			"description": {
+				Types: []string{"string"},
+			},
+			"created_at": DateTime.Schema(),
+			"updated_at": DateTime.Schema(),
+			"options":    Optional(ArrayOf(CustomFieldOptionV1.Schema())),
+		},
+	}
+
+}
+
+func (customFieldV2) Serialize(input client.CustomFieldV2, customFieldOptions []client.CustomFieldOptionV1) map[string]any {
+	var options []map[string]any
```

Comment:
I would personally go for not storing these on the `custom_fields` table, but putting them on their own table

Reply:
Yeah I've gone back and forth on this one - I kind of agree, as this data is also normalized into the incidents table (in this kind of format) but this is more a stream of the original data.

Will split it into 2 streams.